Incopo/Controller/PostDetailViewController.swift

```diff
@@ -138,36 +55,125 @@ class PostDetailViewController: UIViewController {
+            postPictureImageView.heightAnchor.constraint(equalTo: headerView.widthAnchor, multiplier: 9.0/16.0)
+        ])
+    }
+    
+    // MARK: - Table view data source
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 2 + post.comments.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        switch indexPath.row {
+        
+        case 0:
+            guard let imageCell = tableView.dequeueReusableCell(withIdentifier: CellConstants.imageCell, for: indexPath)
+                    as? ImageCell else {
+                return UITableViewCell()
+            }
+            
+            downloadImage(imageURL: post.imageURL) { (image) in
+                imageCell.postImageView.image = image
+            }
+            
+            return imageCell
+        
+        case 1:
+            guard let postDetailCell = tableView.dequeueReusableCell(withIdentifier: CellConstants.postDetailCell, for: indexPath)
+                    as? PostDetailCell else {
+                return UITableViewCell()
+            }
+            configurePostCellData(postDetailCell: postDetailCell)
+            return postDetailCell
+            
+        default:
+            guard let commentCell = tableView.dequeueReusableCell(withIdentifier: CellConstants.commentCell, for: indexPath)
+                    as? CommentCell else {
+                return UITableViewCell()
+            }
+
+            let commentsOwners = Array(post.comments.keys)
+            let currentCommentOwner = commentsOwners[indexPath.row - 2]
+            let commentsContent = Array(post.comments.values)
+            let currentCommentMessage = commentsContent[indexPath.row - 2]
 
-extension PostDetailViewController: UITableViewDelegate {
+            commentCell.nameLabel.text = currentCommentOwner
+            commentCell.commentTextView.text = currentCommentMessage
+            return commentCell
+        }
+    }
     
-    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
     }
-}
-
-extension PostDetailViewController: UITableViewDataSource {
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return post.numberOfComments
+    func configurePostCellData(postDetailCell: PostDetailCell) {
+        postDetailCell.postContentTextView.isEditable = false
+        
+        postDetailCell.titleLabel.text = post.poemTitle
+        postDetailCell.postContentTextView.text = post.poemText
+        fetchAuthor(authorID: post.authorID) {
+            postDetailCell.byAuthorLabel.text = "by ".localized + self.authorName
```

Comment:
it's a little out of scope and we can not change this, but a localized format is better for some languages.
i mean something like "by %1@".localized because this way some languages can translate it to "%1@ is the author"